src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/LibGit2SharpHelpers.cs

```diff
@@ -38,16 +37,23 @@ internal static void SafeCheckout(Repository repo, string commit, CheckoutOption
                                   || e is NameConflictException
                                   || e is LibGit2SharpException)
         {
-            log.LogInformation($"Checkout of {repo.Info.WorkingDirectory} at {commit} failed, fetching before attempting individual files.");
+            log.LogInformation($"Checkout of {repo.Info.WorkingDirectory} at {commit} failed: {e}");
+            log.LogInformation($"Fetching before attempting individual files.");
+
             FetchRepo(repo, log);
+
             try
             {
                 log.LogDebug($"Post-fetch, trying to checkout {repo.Info.WorkingDirectory} at {commit} again");
                 Commands.Checkout(repo, commit, options);
             }
-            catch
+            catch (Exception ex)
             {
-                log.LogWarning($"Couldn't check out one or more files, possibly due to path length limitations ({e.ToString()})." +
+                var isDueToPathLength = ex is InvalidSpecificationException
+                    || ex is NameConflictException
+                    || ex is LibGit2SharpException;
+
+                log.LogWarning($"Couldn't check out one or more files{(isDueToPathLength ? ", possibly due to path length limitations" : "")} ({ex})." +
```

Comment:
The logic is fine, but why is this only warning? What happens when the SafeCheckoutByIndividualFiles checks out the files then? I don't see any return codes there, is the error propagated as an exception?

Also, this same logic is used below, it'd be good to refactor this error handling into a wrapper method / delegate.

Reply:
Well.. if the problem was the long path and we manage to checkout files safely afterwards, we have succeeded and no error needs to be propagated, right?

Reply:
I will factor it out

Reply:
Actually, this would be hard to do. I catch all in the first part and only some in the other to let the exception through

Reply:
Discussed over temas